app/consolidated-view-poc/react/app/LinkedContacts.js

```diff
@@ -0,0 +1,368 @@
+import { html } from 'htm/react'
+import { useEffect, useRef, useState } from 'react'
+import { GET_AUTHENTICATE_QUESTIONS, GET_BUSINESS_CUSTOMERS, GET_CUSTOMER } from './queries.js'
+import { useLazyQuery, useQuery } from './useQuery.js'
+import { useSearch } from './useSearch.js'
+
+export function LinkedContacts({ sbi, email, preloaded }) {
+  // Load list of business customers
+  const { data: businessCustomers, loading: loadingBusinessCustomers } = useQuery(
+    GET_BUSINESS_CUSTOMERS,
+    {
+      variables: { sbi },
+      headers: { email },
+      preloaded: preloaded.businessCustomers
+    }
+  )
+
+  const { search: searchBusinessCustomers, results: businessCustomersSearchResults } = useSearch(
+    businessCustomers?.business?.customers,
+    {
+      idField: 'crn',
+      fields: ['firstName', 'lastName', 'crn'],
+      storeFields: ['firstName', 'lastName', 'crn']
+    }
+  )
+
+  // Create `getCustomer` query
+  const [getCustomer, { data: selectedCustomer, loading: loadingSelectedCustomers }] = useLazyQuery(
+    GET_CUSTOMER,
+    {
+      headers: { email },
+      preloaded: preloaded.selectedCustomer
+    }
+  )
+
+  // Execute `getCustomer` to get first customer in list
+  useEffect(() => {
+    if (businessCustomers?.business?.customers[0].crn) {
+      getCustomer({ crn: businessCustomers?.business?.customers[0].crn, sbi })
+    }
+  }, [businessCustomers])
+
+  // Authenticate questions
+  const [
+    getAuthenticationQuestions,
+    { data: authenticationQuestions, loading: loadingAuthenticationQuestions }
+  ] = useLazyQuery(GET_AUTHENTICATE_QUESTIONS, {
+    headers: { email }
+  })
+
+  const [showAuthenticationQuestions, setShowAuthenticationQuestions] = useState(false)
+
+  useEffect(() => {
+    if (
+      (showAuthenticationQuestions && !authenticationQuestions) ||
+      (showAuthenticationQuestions &&
+        authenticationQuestions?.customer?.crn !== selectedCustomer?.customer?.crn)
```

Comment:
You've got operations with **Associativity** _and_ **Commutativity** here, so you can use the **Distributive law**:
```suggestion
      showAuthenticationQuestions && (!authenticationQuestions ||
        (authenticationQuestions?.customer?.crn !== selectedCustomer?.customer?.crn))
```
Good old A-level Maths 😅